src/controllers/article.controller.js

```diff
@@ -150,6 +153,121 @@ class ArticleController {
+   * @param {*} response -- Response object
+   * @returns {json} -- Returns a json object
+   * @memberof ArticleController
+   */
+  static async createArticle(request, response) {
+    const { id: authorId } = request.currentUser;
+    const { title } = request.body;
+    const createDate = new Date();
+    try {
+      const article = await BaseRepository.create(db.Article, {
+        ...request.body,
+        slug: slug(
+          `${title}-${crypto.randomBytes(12).toString('base64')}`
+        ).toLowerCase(),
+        authorId,
+        createdAt: createDate
+      });
+      return responseGenerator.sendSuccess(
+        response,
+        201,
+        article,
+        'Artilcle successfully created'
+      );
+    } catch (error) {
+      return responseGenerator.sendError(response, 500, error.message);
+    }
+  }
+
+  /**
+   *
+   *
+   * @static
+   * @param {*} request - - express request parameter
+   * @param {*} response - - express response parameter
+   * @returns {object} - - article object
+   * @memberof ArticleController
+   */
+  static async fetchSpecificArticle(request, response) {
+    const { article } = request;
+    return responseGenerator.sendSuccess(response, 200, article);
+  }
+
+  /**
+   *
+   *
+   * @static
+   * @param {*} request - - express request parameter
+   * @param {*} response - - express response parameter
+   * @returns  {object} - - article object
+   * @memberof ArticleController
+   */
+  static async deleteArticle(request, response) {
+    const { articleId } = request.params;
+    try {
+      await BaseRepository.remove(db.Article, { id: articleId });
+      return responseGenerator.sendSuccess(
+        response,
+        200,
+        null,
+        'Article successfully deleted'
+      );
+    } catch (error) {
+      return responseGenerator.sendError(response, 500, error.message);
+    }
+  }
+
+  /**
+   *
+   *
+   * @static
+   * @param {*} request - - express request parameter
+   * @param {*} response - - express response parameter
+   * @returns {object} - - article object
+   * @memberof ArticleController
+   */
+  static async updateOneArticle(request, response) {
+    const { articleId } = request.params;
+    const { id: authorId } = request.currentUser;
+    const { title, description, body, image } = request.body;
+    const newUpdateDate = new Date();
+
+    try {
+      const article = await BaseRepository.findOneByField(db.Article, {
+        id: articleId,
+        authorId
+      });
+      const updatedArticle = await BaseRepository.updateField(article, {
+        title,
+        description,
+        body,
+        image,
+        updatedAt: newUpdateDate
+      });
+      return article
+        ? responseGenerator.sendSuccess(
+            response,
+            200,
+            updatedArticle,
+            'Article successfully updated'
+          )
+        : responseGenerator.sendError(
+            response,
+            404,
```

Comment:
This check should be done immediately after the `findOneByField`

Reply:
Thank you for this feedback.
This also relates to the previous comment.

Here, is an update operation that is meant to happen on request.
On `Line 244`, the request is done with the `articleId` set on the `request.params`.
If the article is found, `Line 248` is executed and completed on `Line 254`.
`Line 255` starts a tenary operation which checks for the truthy or falsy value of the request made.

While working on this with Princewill during a pair-programming session, we discovered that this implementation gave better code coverage than doing it the way to have mentioned here.

Again this implementation first tested for the expected activity of the implementation and then gives a response based on the result of the operation.